### PR TITLE
5.1 inclusion into minor

### PR DESCRIPTION
**Pull request by obuisard**

### CODE REVIEW

**Comment by zero-24** [comment]
I would say we should also allow updates from 4.4-dev and 5.0-dev to 5.1-dev right?

**Comment by HLeithner** [comment]
can we directly restructure this into understandable directories and file names please

**Comment by obuisard** [comment]
> can we directly restructure this into understandable directories and file names please

You mean like you did with next.xml for instance? Maybe should be done in a separate PR.

**Comment by HLeithner** [comment]
splitting first makes more sense, to make it clear what should happen

**Comment by obuisard** [comment]
> splitting first makes more sense, to make it clear what should happen

I have split the files. I was able to spot a few mistakes (at least, I thought they were mistakes...).
I hope this is what you were looking for.

**Comment by HLeithner** [comment]
are your spotted mistakes related to 4.0.4 and lower versions?

**Comment by obuisard** [comment]
> are your spotted mistakes related to 4.0.4 and lower versions?

In list_test:

```
<extension name="Joomla" element="joomla" type="file" version="4.4.0-rc" targetplatformversion="3.10.12" detailsurl="https://update.joomla.org/core/test/extension_test.xml" />
	<extension name="Joomla" element="joomla" type="file" version="4.4.0-rc" targetplatformversion="3.10.13" detailsurl="https://update.joomla.org/core/test/extension_test.xml" />
```

did not seem right to me because nothing mentioned in extension_test.xml

**Comment by obuisard** [comment]
I also make sure 4.0.4 is treated differently:

`<extension name="Joomla" element="joomla" type="file" version="4.0.4" targetplatformversion="4.0" detailsurl="https://update.joomla.org/core/j4/default.xml" />`

goes to the default.xml file rather that patch.xml for instance, because we always need to install it first on 4.0 systems before anything else.

File: www/core/nightlies/next_minor_extension.xml [hunk 1]
```diff
@@ -19,81 +19,4 @@
...
-		</downloads>
-		<tags>
-			<tag>stable</tag>
-		</tags>
-		<supported_databases mysql="8.0.13" mariadb="10.4" postgresql="12.0" />
-		<php_minimum>8.1.0</php_minimum>
-		<maintainer>Joomla! Production Department</maintainer>
-		<maintainerurl>https://www.joomla.org</maintainerurl>
-		<targetplatform name="joomla" version="5.[0]" />
-	</update>
```
Comment (HLeithner) [inline www/core/nightlies/next_minor_extension.xml:98]: you removed the special handling for 4.0.x series at least I'm not sure what you are doing with it?

did you changed it to j4/default.xml?

Comment (obuisard) [reply www/core/nightlies/next_minor_extension.xml:98]: It's in minor.xml

Comment (obuisard) [reply www/core/nightlies/next_minor_extension.xml:98]: Yes, all handling of 4.0 is handled in j4/default.xml

**Comment by tecpromotion** [comment]
@obuisard obsolete?

**Comment by obuisard** [comment]
> @obuisard obsolete?

Yes, now it is.

**Pull request closed without merging**